data/remote/PostServiceImpl.kt

```diff
@@ -0,0 +1,64 @@
+package com.example.revontulet.data.remote
+
+import io.ktor.client.HttpClient
+import io.ktor.client.call.body
+import io.ktor.client.request.post
+import io.ktor.client.request.setBody
+import io.ktor.http.ContentType
+import io.ktor.http.contentType
+
+ /*
+  * KTOR service for sending post requests to the server
+  * Everything is in try catch construction to prevent system from crashing
+  */
+
+class PostServiceImpl(
+    private val client: HttpClient
+) : PostService {
+    // Post single entry to server
+    override suspend fun postGeoSignal(postRequest: PostRequest): PostResponse? {
+        return try {
+            val response = client.post(ApiConstants.BASE_URL) {
+                contentType(ContentType.Application.Json)
+                setBody(postRequest)
+            }
+
+            if (response.status.value in 200..299) {
+                response.body()
+            } else {
+                PostResponse(
+                    success = false,
+                    message = "Something went wrong, status code: ${response.status.value}"
+                )
+            }
```

Comment:
Het gebruik van `when` over if/else bij getal ranges zijn effecienter. Ook is het fijner om de variabele minder vaak te hoeven gebruiken dus je zou het kunnen gebruiken in een .let{} constructie.

```suggestion
            response.status.value.let {
              when (it) {
                in 200..299 -> response.body()
                else -> PostResponse(
                    success = false,
                    message = "Something went wrong, status code: ${it}"
                )
              }
            }
```